src/main/java/com/ericsson/esj/helpers/JenkinsXmlData.java

```diff
@@ -0,0 +1,224 @@
+        String hudsonGroovyKey = "hudson.plugins.groovy.Groovy plugin='groovy@2.1'";
+        boolean hasKeyGroovy = builders.has(hudsonGroovyKey);
+
+        if (!hasKeyGroovy) {
+            JSONArray hudsonTasksGroovy = new JSONArray();
+            builders.put(hudsonGroovyKey, hudsonTasksGroovy);
+        }
+
+        JSONObject newGroovyCommand = BuildGroovyObject(script);
+
+        builders.getJSONArray(hudsonGroovyKey).put(newGroovyCommand);
+    }
+
+    /**
+     * This function adds a parameter key to the job data, the user must specify
+     * what type the parameter will receive, currently only String.class and
+     * boolean.class is supported.
+     *
+     * @param key
+     *            :: The parameter key
+     * @param defaultValue
+     *            :: The default value if parameter is empty
+     * @param description
+     *            :: Description
+     * @param trim
+     *            :: Trim white spaces
+     * @throws Exception
+     */
+    public void addBuildParameter(String key, String defaultValue, String description, boolean trim) throws Exception {
+        String parametertypeKey = "hudson.model.StringParameterDefinition";
+
+        validatePropertiesObject(parametertypeKey);
+
+        JSONObject param = new JSONObject();
+        param.put("name", key);
+        param.put("description", description);
+        param.put("defaultValue", defaultValue);
+        param.put("trim", false);
+
+        properties.getJSONObject(HUDSON_PARAMETERS_DEFENITION_KEY).getJSONObject(PARAMETER_DEFENITION_KEY)
+                .getJSONArray(parametertypeKey).put(param);
+    }
+
+    /**
+     * This function adds a parameter key to the job data, the user must specify
+     * what type the parameter will receive, currently only String.class and
+     * boolean.class is supported. No default value, no description and trim
+     * false
+     *
+     * @param key
+     *            :: The parameter key
+     * @throws Exception
+     */
+    public void addBuildParameter(String key) throws Exception {
+        addBuildParameter(key, "", "", false);
+    }
+
+    /**
+     * Ensures the XML properties is added correctly, should be
+     * <hudson.model.ParametersDefinitionProperty> <parameterDefinitions>
+     * <our-input-parameter-typ> </our-input-parameter-typ>
+     * </parameterDefinitions> </hudson.model.ParametersDefinitionProperty>
+     *
+     * @param parametertypeKey
+     */
+    private void validatePropertiesObject(String parametertypeKey) {
+        boolean hasHudsonParametersDefinitionKey = properties.has(HUDSON_PARAMETERS_DEFENITION_KEY);
+        if (!hasHudsonParametersDefinitionKey) {
+            JSONObject parametersDefinitionProperty = new JSONObject();
+            properties.put(HUDSON_PARAMETERS_DEFENITION_KEY, parametersDefinitionProperty);
+        }
+
+        boolean hasParameterDefinitions = properties.getJSONObject(HUDSON_PARAMETERS_DEFENITION_KEY)
```

Comment:
HUDSON_PARAMETERS_DEFENITION_KEY ->HUDSON_PARAMETERS_DEFINITION_KEY.  Spelling mistake, same with the other constants :)

Reply:
Done